https://bugs.gentoo.org/900286
https://github.com/lldpd/lldpd/commit/01aee0f3601c60e570aeb9040c036c83d053cc5a

From 01aee0f3601c60e570aeb9040c036c83d053cc5a Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Thu, 8 Dec 2022 13:01:07 +0100
Subject: [PATCH] m4/progname.m4: Include <stdio.h> for printf in
 lldp_CHECK___PROGNAME

Otherwise the checks always fails with a compiler that does not
support implict function declarations.
--- a/m4/progname.m4
+++ b/m4/progname.m4
@@ -4,7 +4,7 @@
 AC_DEFUN([lldp_CHECK___PROGNAME],[
   AC_CACHE_CHECK([whether libc defines __progname], lldp_cv_check___progname, [
     AC_LINK_IFELSE([AC_LANG_PROGRAM(
-                     [[]],
+                     [[#include<stdio.h>]],
                      [[ extern char *__progname; printf("%s", __progname); ]])],
                      [ lldp_cv_check___progname="yes" ],
                      [ lldp_cv_check___progname="no" ])

